x-pack/plugins/ml/server/models/data_recognizer/modules/apm_transaction/ml/datafeed_apm_metrics.json

```diff
@@ -0,0 +1,95 @@
+{
+  "job_id": "JOB_ID",
+  "indices": [
+    "INDEX_PATTERN_NAME"
+  ],
+  "chunking_config" : {
+    "mode" : "off"
+  },
+  "query": {
```

Comment:
@dgieselaar this change looks good when creating the jobs from within the ML app, but just wondering if you also need to add them in the overrides you are supplying for the datafeed when creating the jobs from the APM app in `x-pack/plugins/apm/server/lib/anomaly_detection/create_anomaly_detection_jobs.ts`? Currently that overrides the module query to only filter on the `service.environment`.

Reply:
Do those not get merged?

Reply:
Just to close the loop on this one, when the job is created from the APM UI, a filter for the service environment is added to the query defined in the module, as done by @dgieselaar in https://github.com/elastic/kibana/pull/111957/commits/1dc720eb1797e764c38691644cd9161f34681438.

So for example, the datafeed query for a job created for the `staging` environment will end up as:

```
    "query": {
      "bool": {
        "filter": [
          {
            "term": {
              "processor.event": "metric"
            }
          },
          {
            "term": {
              "metricset.name": "transaction"
            }
          },
          {
            "term": {
              "service.environment": "staging"
            }
          }
        ]
      }
    },
```

Reply:
What's the purpose of merging the two? Can't we just have the full definition in a single place -  preferably in the APM codebase.